internal/provisioner/objects/service/service.go

```diff
@@ -196,10 +196,11 @@ func DesiredEnvoyService(contour *model.Contour) *corev1.Service {
 			Labels:      model.CommonLabels(contour),
 		},
 		Spec: corev1.ServiceSpec{
-			Ports:           ports,
-			Selector:        dataplane.EnvoyPodSelector(contour).MatchLabels,
-			SessionAffinity: corev1.ServiceAffinityNone,
-			LoadBalancerIP:  contour.Spec.NetworkPublishing.Envoy.LoadBalancer.LoadBalancerIP,
+			Ports:                 ports,
+			Selector:              dataplane.EnvoyPodSelector(contour).MatchLabels,
+			SessionAffinity:       corev1.ServiceAffinityNone,
+			LoadBalancerIP:        contour.Spec.NetworkPublishing.Envoy.LoadBalancer.LoadBalancerIP,
+			ExternalTrafficPolicy: corev1.ServiceExternalTrafficPolicyTypeLocal,
```

Comment:
We need to drop this line, since we can only set this field if the Service type is NodePort or LoadBalancer. It's already being set in those cases on L254.